### PR TITLE
estimatePoseBoard: Add flags for choosing initial guess and solvepnp methods

### PR DESCRIPTION
**Pull request by lorenzolightsgdwarf**
Based on the cv::solvepnp signature, these small changes allow 

- to use rvec and tvec as initial guess for the iterative solvers

- to select the solving method, such as EPNP, Iterative etc.

### CODE REVIEW

File: modules/aruco/include/opencv2/aruco.hpp [hunk 1]
```diff
@@ -336,7 +336,7 @@ class CV_EXPORTS_W GridBoard : public Board {
  */
 CV_EXPORTS_W int estimatePoseBoard(InputArrayOfArrays corners, InputArray ids, Ptr<Board> &board,
```
Comment (StevenPuttemans) [inline modules/aruco/include/opencv2/aruco.hpp]: You should make sure that the extra defined parameters here also get assigned a `@param` explanation above!

**Comment by lorenzolightsgdwarf** [comment]
Hi, I think I implemented your comment. But the build-bot doesn't build the doc. Do you know how to spot the problem? I checked the white spaces  and it compiles for me.
Thanks

**Comment by mshabunin** [comment]
@lorenzolightsgdwarf  , please remove trailing whitespaces:
```
modules/aruco/include/opencv2/aruco.hpp:327: trailing whitespace.
+ * @param useExtrinsicGuess Parameter used for SOLVEPNP_ITERATIVE. If true (1), the function uses 
modules/aruco/include/opencv2/aruco.hpp:347: trailing whitespace.
+ * 
```

**Comment by paroj** [comment]
I would prefer the approach taken in #748 to this.

**Comment by lorenzolightsgdwarf** [comment]
Great. I close this pull request

**Pull request closed without merging**